README.md

```diff
@@ -1,28 +1,78 @@
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          github-token: ${{ steps.generate_token.outputs.token }}
+
 ```
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
+
+This action uses the [listRepoSecrets endpoint](https://docs.github.com/en/rest/reference/actions#list-repository-secrets), your GitHub App must have the `secrets` repository permission to use this endpoint.
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
+
+### uses: tibdex/github-app-token@v1
+
+**Required**
+This step is used when `secrets.GITHUB_TOKEN`'s limitations are too restrictive and a personal access token is not suitable.
+
+The default `secrets.GITHUB_TOKEN` has limited permissions [click here for more information](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), this step is generating a new token. `APP_ID` and `PRIVATE_KEY` are required inputs, which have been generated by installing a GitHub App on the previous step
+
+```yml
+- name: Generate token
+  id: generate_token
+  uses: tibdex/github-app-token@v1
+  with:
+    app_id: ${{ secrets.APP_ID }}
+    private_key: ${{ secrets.PRIVATE_KEY }}
+```
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
+
+### uses: actions/checkout@v3
+
+**Required** This action checks-out your repository under `$GITHUB_WORKSPACE`. For information check [actions/checkout](https://github.com/actions/checkout). This step is required because under the hood `detect-unused-secrets` is executing a script to grab the secret names used in `.github/workflows`.
+
 ### `github-token`
 
-**Required** A GitHub token. For additional information read this guide --> [#about-the-github_token-secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret)
+**Required** [The generated token](#uses-tibdexgithub-app-tokenv1) or your personal access token with repo scope.
```

Comment:
the scope is wrong

Reply:
The repo scope is required for this action to run with a PAT, do you mean the naming? This is named scope in Github.
<img width="653" alt="Screenshot 2022-03-09 at 11 56 52" src="https://user-images.githubusercontent.com/32770804/157428199-c8e82c11-0bc8-47d1-a89f-045c0e518766.png">

Reply:
Ah no you're actually right again. The repo scope is required for the PAT. Instead of specifying it in detail here, why don't we just point the users to the docs? https://docs.github.com/en/rest/reference/actions#list-repository-secrets